packages/snaps-controllers/src/services/AbstractExecutionService.ts

```diff
@@ -399,7 +397,7 @@ export abstract class AbstractExecutionService<WorkerType>
           handler,
           request,
           target: snapId,
-        },
+        } as Record<string, Json>,
```

Comment:
I'm guessing `params` is causing grief here? if so, could we move the type assertion to `params`?

Reply:
It is in fact `request` that is typed as `Record<string, Json>` but should be a `JsonRpcRequest` by the time we get here. I have moved the cast.